urlconnection/src/main/java/zipkin/reporter/urlconnection/URLConnectionReporter.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright 2016 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.reporter.urlconnection;
+
+import com.google.auto.value.AutoValue;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.Executor;
+import java.util.zip.GZIPOutputStream;
+import zipkin.Codec;
+import zipkin.Span;
+import zipkin.storage.AsyncSpanConsumer;
+import zipkin.storage.Callback;
+
+/**
+ * Reports spans to Zipkin, using its {@code POST /spans} endpoint.
+ */
+@AutoValue
+public abstract class URLConnectionReporter implements AsyncSpanConsumer {
+
+  /**
+   * Returns the POST URL for zipkin's <a href="http://zipkin.io/zipkin-api/#/">v1 api</a>, usually
+   * "http://zipkinhost:9411/api/v1/spans".
+   *
+   * <p>This is exposed for those who wish to set this dynamically.
+   */
+  // @FunctionalInterface except library compatible w/ JRE 7
+  public interface PostURL {
```

Comment:
I was thinking more that it was wasteful to incur the cost of recreating the same URL for every request if there was no need for it to be dynamic. But thinking about it some more: (1) this only happens once on a period basis - e.g. once per second, and (2) it's cheap to create a URL anyway. So having the possibility for a dynamic URL doesn't really cost much other than some minor code maintenance hassle. 👍